docs/source/tutorial_docker.md

```diff
@@ -0,0 +1,84 @@
+This image comes with pre-installed ParlAI, along with some extra packages often required for using it.
+You can find it under Packages in our main Github repostiry ([link](https://github.com/orgs/facebookresearch/packages?repo_name=ParlAI)).
+Using this image, you can run ParlAI *anywhere* that you can run Docker, without having to install ParlAI or worry about its dependancies.
+
+## Pulling the image
+To try the latest version of the package simply run
+```console
+docker run ghcr.io/facebookresearch/parlai:latest
+```
+The default command in the Docker image is `parlai party`.
+With that, you will see the dancing parrot after you run the image on its default command (as we did here).
+
+### Running single ParlAI command
+You can override the default image command to any other ParlAI (in fact container OS) command by appending it after the above command:
+```console
+docker run ghcr.io/facebookresearch/parlai:latest \
+<PARLAI_COMMAND>
+```
+Here, `<PARLAI_COMMAND>` is the command that you want to run.
+For example, if you want to run `parlai display_data -t wizard_of_wikipedia`,
+then you will run
+```console
+docker run ghcr.io/facebookresearch/parlai:latest \
+parlai display_data -t wizard_of_wikipedia
+```
+
+### Interactive ParlAI shell
+The use case explained above only runs a single ParlAI command.
+A better usage can be running an interactive terminal with the environment provided by this image.
+The following command will give you a shell inside the container:
+```console
+docker run -it ghcr.io/facebookresearch/parlai:latest bash
+```
+Once in this terminal you can run any ParlAI command and have interactions with ParlAI as you have it installed on your machine.
+After you are done you may exit this terminal as usual (`exit` command or <kbd>Ctrl</kbd> + <kbd>D</kbd>).
+
+## Runtime resources
+Running the interactive version of the image, as we did above, will give you the default runtime resources in your container
+(if you have Docker dashboard installed, you can configure this via `Preferences` setting).
+Depending on what you want to do, this may not be enough.
+For example, most likely the default setting will not include the GPU resources you may need;
+so, if you are inside the terminal of the image from the command abover,
+trying `python -c 'import torch;print(torch.cuda.is_available())'` will return `False`, regardless of the number of GPUs you have on your machine.
+
+The solution is to add extra resources (eg, RAM, GPU etc.) to your runtime environment.
+For example, the following command will give your runtime environment 2 GPUs
+(the ones with 0 and 3 ids) if your machine has them at its disposal:
+```console
+docker run -it --gpus 0,3 ghcr.io/facebookresearch/parlai:latest bash
+```
+For full furthure on setting other resources (CPU, memory, etc.) consult [Docker documentation](https://docs.docker.com/config/containers/resource_constraints/).
```

Comment:
`for full details`?

Reply:
Haha, this one is really funny. I think most of these happen when I edit without going back and reading what I just edited.